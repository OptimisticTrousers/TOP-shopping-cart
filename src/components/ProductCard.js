import { Link } from "react-router-dom"
import star from '../assets/star.svg'

export default function ProductCard({image, rating, title, price}) {
    return (
        <div className="product" data-testid="product-card">
            <div className="product-image">
                <img src={image} alt={title}/>
            </div>
            <div className="product-rating">
                <p className="product-review"><img className="star" src={star} alt="yellow star"/> {rating.rate} ({rating.count} Reviews)</p>
            </div>
            <div className="product-detail">
                <h2 className="product-title">{title}</h2>
                <p className="price-name">Price:</p>
                <p className="product-price">${price}</p>
            </div>
        </div>
    )
}