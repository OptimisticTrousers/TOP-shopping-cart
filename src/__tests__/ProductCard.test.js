import {render, screen} from '@testing-library/react'
import ProductCard from '../components/ProductCard'

describe("ProductCard", () => {
    beforeEach(() => {

        fetch.mockClear()

        fetch.mockResponseOnce(JSON.stringify())
    })

    const product = { 
        title: "Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops", 
        price: 109.95, 
        image: "https://fakestoreapi.com/img/81fPKd-2AYL._AC_SL1500_.jpg", 
        rating: { rate: 3.9, count: 120 }
    }

    it("snapshot", () => {

        const {asFragment} = render(<ProductCard image={product.image} rating={product.rating} title={product.title} price={product.price} />)

        expect(asFragment()).toMatchSnapshot()
    })
    it("ProductCard renders correctly when given props", () => {

        render(<ProductCard image={product.image} rating={product.rating} title={product.title} price={product.price} />)

        const productTitle = screen.queryByText(product.title)
        const productImage = screen.queryByAltText(product.title)
        const productRating = screen.queryByTestId('product-rating')
        const productPrice = screen.getByText(/\$/i)

        expect(productTitle.textContent).toBe(product.title)
        expect(productImage.src).toBe(product.image)
        expect(productRating.textContent).toBe(` ${product.rating.rate} (${product.rating.count} Reviews)`)
        expect(productPrice.textContent).toBe(`$${product.price}`)

    })
})