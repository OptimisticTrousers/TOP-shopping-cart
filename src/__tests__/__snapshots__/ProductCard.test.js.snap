// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard snapshot 1`] = `
<DocumentFragment>
  <div
    class="product"
    data-testid="product-card"
  >
    <div
      class="product-image"
    >
      <img
        alt="Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops"
        src="https://fakestoreapi.com/img/81fPKd-2AYL._AC_SL1500_.jpg"
      />
    </div>
    <div
      class="product-rating"
    >
      <p
        class="product-review"
        data-testid="product-rating"
      >
        <img
          alt="yellow-star"
          class="star"
          src="star.svg"
        />
         3.9 (120 Reviews)
      </p>
    </div>
    <div
      class="product-detail"
    >
      <h2
        class="product-title"
      >
        Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops
      </h2>
      <p
        class="price-name"
      >
        Price:
      </p>
      <p
        class="product-price"
      >
        $109.95
      </p>
    </div>
  </div>
</DocumentFragment>
`;
